### ISSUE TITLE
Suggested Documentation/FAQ update

### ISSUE THREAD

**Issue by rugubara** [zfsonlinux/zfs#6274]
### System information

Type                                | Version/Name
  ---                                  |     --- 
Distribution Name       |  Gentoo
Distribution Version    |  current
Linux Kernel                 |  4.11.7
Architecture                 |  amd64
ZFS Version                  |  0.7.0-rc3
SPL Version                  |  0.7.0-rc4_4_gac48361

### Describe the problem you're observing
ZoL has certain "peculiarities" that an average user would not expect from a modern Linux filesystem in 2017. 

- abscense of TRIM support for vdevs/underlying media. This brings an SSD to a situation when it must trim before every write very soon. it takes 2-4 secs per write then!
- abscence of suspend to RAM/Disk support. It is important if ZFS is used to power a laptop/desktop rather than a server

I would like to suggest that these things are mentioned on FAQ/Installation page to save potential users some frustration.

**Comment by bunder2015** [comment]
TRIM is in PR stage, see #5925

**Comment by gmelikov** [comment]
> abscense of TRIM support ... must trim before every write very soon
ZFS is a CoW FS, it doesn't write one data to same place.

**Comment by rugubara** [comment]
> TRIM is in PR stage, see #5925
Indeed. However there is a comment that it'll not be merged into 0.7.0. Anyone on release version will not be able to trim their SSDs. I suffered from the abscense of trim myself recently when  SSD worked slower than a SATA HDD. 
Before you comment on "harmlessnes" of unrimmed SSD, I suggest you put your / , /var and /usr on such a disk and EXPERIENCE a saturated SSD when +50% of your writes take more than 2 secs also holding the reads in the queue.
I had to switch to dweeezil's ntrim-next-2 branch to heal my system back to life. 
I strongly believe that such advice/information must be present to a person attempting a migration to ZFS.

**Comment by gmelikov** [comment]
@rugubara  how much free space did you have on this SSD?

**Comment by rugubara** [comment]
158G  used 75.3G free

**Comment by bunder2015** [comment]
> Before you comment on "harmlessnes" of untrimmed SSD, I suggest you put your / , /var and /usr on such a disk and EXPERIENCE a saturated SSD when +50% of your writes take more than 2 secs also holding the reads in the queue.

I have several SSD pools, no problems here.

**Comment by gmelikov** [comment]
I have SSD, ZFS on LUKS, no problems too.
@rugubara can you give us pool geometry and properties?

**Comment by rugubara** [comment]
```
r9-008cln ~ # zpool status VM
  pool: VM
 state: ONLINE
status: Some supported features are not enabled on the pool. The pool can
        still be used, but some features are unavailable.
action: Enable all features using 'zpool upgrade'. Once this is done,
        the pool may no longer be accessible by software that does not support
        the features. See zpool-features(5) for details.
  scan: none requested
  trim: completed on Tue Jun 27 12:05:27 2017 (after 0h7m)
config:

        NAME        STATE     READ WRITE CKSUM
        VM          ONLINE       0     0     0
          sdb3      ONLINE       0     0     0
          sdb2      ONLINE       0     0     0

errors: No known data errors

r9-008cln ~ # sfdisk -l /dev/sdb
Disk /dev/sdb: 238.5 GiB, 256060514304 bytes, 500118192 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disklabel type: gpt
Disk identifier: A5301C29-F3CD-455F-ADC0-46427A3E173C

Device         Start       End   Sectors   Size Type
/dev/sdb1       2048   8390655   8388608     4G FreeBSD ZFS
/dev/sdb2    8390656 142608383 134217728    64G FreeBSD ZFS
/dev/sdb3  142608384 500118158 357509775 170.5G FreeBSD ZFS

r9-008cln ~ # zfs get all VM
NAME  PROPERTY              VALUE                  SOURCE
VM    type                  filesystem             -
VM    creation              Sun Jun  4 18:56 2017  -
VM    used                  189G                   -
VM    available             46.5G                  -
VM    referenced            20K                    -
VM    compressratio         1.09x                  -
VM    mounted               yes                    -
VM    quota                 none                   default
VM    reservation           none                   default
VM    recordsize            128K                   default
VM    mountpoint            /mnt/VM                local
VM    sharenfs              off                    default
VM    checksum              on                     default
VM    compression           lz4                    local
VM    atime                 off                    local
VM    devices               on                     default
VM    exec                  on                     default
VM    setuid                on                     default
VM    readonly              off                    default
VM    zoned                 off                    default
VM    snapdir               hidden                 default
VM    aclinherit            restricted             default
VM    canmount              on                     default
VM    xattr                 on                     default
VM    copies                1                      default
VM    version               5                      -
VM    utf8only              off                    -
VM    normalization         none                   -
VM    casesensitivity       sensitive              -
VM    vscan                 off                    default
VM    nbmand                off                    default
VM    sharesmb              off                    default
VM    refquota              none                   default
VM    refreservation        none                   default
VM    primarycache          all                    default
VM    secondarycache        all                    default
VM    usedbysnapshots       6K                     -
VM    usedbydataset         20K                    -
VM    usedbychildren        189G                   -
VM    usedbyrefreservation  0                      -
VM    logbias               latency                default
VM    dedup                 off                    default
VM    mlslabel              none                   default
VM    sync                  standard               default
VM    dnodesize             legacy                 default
VM    refcompressratio      1.00x                  -
VM    written               0                      -
VM    logicalused           180G                   -
VM    logicalreferenced     10K                    -
VM    filesystem_limit      none                   default
VM    snapshot_limit        none                   default
VM    filesystem_count      none                   default
VM    snapshot_count        none                   default
VM    snapdev               hidden                 default
VM    acltype               off                    default
VM    context               none                   default
VM    fscontext             none                   default
VM    defcontext            none                   default
VM    rootcontext           none                   default
VM    relatime              off                    default
VM    redundant_metadata    all                    default
VM    overlay               off                    default

Jun 27 11:49:42 scsi 4:0:0:0: Direct-Access     ATA      TS256GMTS400     4B   PQ: 0 ANSI: 5
Jun 27 11:49:42 r9-008cln kernel: sd 4:0:0:0: [sdb] 500118192 512-byte logical blocks: (256 GB/238 GiB)
Jun 27 11:49:42 r9-008cln kernel: sd 4:0:0:0: Attached scsi generic sg1 type 0
Jun 27 11:49:42 r9-008cln kernel: sd 4:0:0:0: [sdb] Write Protect is off
Jun 27 11:49:42 r9-008cln kernel: sd 4:0:0:0: [sdb] Mode Sense: 00 3a 00 00
Jun 27 11:49:42 r9-008cln kernel: sd 4:0:0:0: [sdb] Write cache: enabled, read cache: enabled, doesn't support DPO or FUA
```

**Comment by rugubara** [comment]
```
r9-008cln ~ # zpool get all VM
NAME  PROPERTY                    VALUE                       SOURCE
VM    size                        234G                        -
VM    capacity                    67%                         -
VM    altroot                     -                           default
VM    health                      ONLINE                      -
VM    guid                        7491683219797843185         -
VM    version                     -                           default
VM    bootfs                      VM/system                   local
VM    delegation                  on                          default
VM    autoreplace                 off                         default
VM    cachefile                   -                           default
VM    failmode                    wait                        default
VM    listsnapshots               off                         default
VM    autoexpand                  off                         default
VM    dedupditto                  0                           default
VM    dedupratio                  1.25x                       -
VM    free                        75.3G                       -
VM    allocated                   158G                        -
VM    readonly                    off                         -
VM    ashift                      0                           default
VM    comment                     -                           default
VM    expandsize                  -                           -
VM    freeing                     0                           -
VM    fragmentation               72%                         -
VM    leaked                      0                           -
VM    forcetrim                   off                         default
VM    autotrim                    on                          local
VM    feature@async_destroy       enabled                     local
VM    feature@empty_bpobj         active                      local
VM    feature@lz4_compress        active                      local
VM    feature@spacemap_histogram  active                      local
VM    feature@enabled_txg         active                      local
VM    feature@hole_birth          active                      local
VM    feature@extensible_dataset  active                      local
VM    feature@embedded_data       active                      local
VM    feature@bookmarks           enabled                     local
VM    feature@filesystem_limits   active                      local
VM    feature@large_blocks        enabled                     local
VM    feature@large_dnode         disabled                    local
VM    feature@sha512              disabled                    local
VM    feature@skein               disabled                    local
VM    feature@edonr               disabled                    local
VM    feature@userobj_accounting  disabled                    local

```

**Comment by bunder2015** [comment]
I'm not sure why you're trying to stripe a single SSD via partitions, you're much better off buying a second SSD and setting up a proper (whole disk) mirror or stripe configuration.

**Comment by rugubara** [comment]
> I'm not sure why you're trying to stripe a single SSD via partitions, you're much better off buying a second SSD and setting up a proper (whole disk) mirror or stripe configuration.

This is historical... sdb2 used to be L2ARC for a HDD pool. Turned out to be quite useless in my usage scenarios. I've added the space back to SSD pool. But we are diverging from my topic.

**Comment by kpande** [comment]
this isn't the proper location for this discussion - try IRC (#zfsonlinux @ Freenode) or [the zfs-discuss mailing list](http://list.zfsonlinux.org). that said; ZFS is geared toward enterprise users and these issues are non-issues. My zfs root systems suspend to RAM and disk (though I use a LUKS external just for this) *and* wake up just fine.

**Comment by s-m-e** [comment]
@kpande In the context of FAQ & documentation - I am researching how to safely do suspend to RAM/disk with ZFS. I have not found an authorative source so far other than issue #260.

> [...] I use a LUKS external just for this [...]

Could you please elaborate on how exactly your configuration looks like?

**Comment by bunder2015** [comment]
S3 (suspend to ram) works just fine for me out of the box, no zfs specific setup required.  S4 (hibernate) might be a little more challenging however.

**Comment by kpande** [comment]
if s2ram doesn't work, then it's a problem with Linux and nothing to do with ZFS.

s2disk needs initrd support, and i'm not sure a) how to implement that in dracut (I looked), or b) how the upstream devs would react to the changes since they could be quite invasive.
